support-modules/acquisition-events/src/main/scala/com/gu/support/acquisitions/ga/GoogleAnalyticsServiceImpl.scala

```diff
@@ -0,0 +1,255 @@
+package com.gu.support.acquisitions.ga
+
+import cats.data.EitherT
+import com.gu.acquisitionsValueCalculatorClient.model.{AcquisitionModel, PrintOptionsModel}
+import com.gu.acquisitionsValueCalculatorClient.service.AnnualisedValueService
+import com.gu.support.acquisitions.ga.GoogleAnalyticsServiceImpl._
+import com.gu.support.acquisitions.ga.models.GAError.{BuildError, NetworkFailure, ResponseUnsuccessful}
+import com.gu.support.acquisitions.ga.models.{ConversionCategory, GAData, GAError}
+import com.gu.support.acquisitions.models.AcquisitionProduct.{Contribution, DigitalSubscription, Paper, RecurringContribution}
+import com.gu.support.acquisitions.models.{AbTest, AcquisitionDataRow, AcquisitionProduct, PrintOptions}
+import com.gu.support.acquisitions.models.PrintProduct._
+import com.gu.support.acquisitions.utils.Retry
+import com.typesafe.scalalogging.LazyLogging
+import okhttp3.{Call, Callback, HttpUrl, OkHttpClient, Request, RequestBody, Response}
+
+import java.io.IOException
+import java.util.UUID
+import scala.concurrent.{ExecutionContext, Future, Promise}
+import scala.util.control.NonFatal
+import scala.util.matching.Regex
+
+object GoogleAnalyticsServiceImpl {
+  val clientIdPattern: Regex = raw"GA\d\.\d\.(\d+\.\d+)".r
+
+  // e.g. GUARDIAN_WEEKLY becomes GuardianWeekly
```

Comment:
this is for where we used to use the `name` field on the thrift models.
E.g. `acquisition.paymentFrequency.name` is `OneOff`

Reply:
I don't understand this.  Do you mean, the thrift models used to magically convert the case, and now you have written a function to repliacate that?

Reply:
Exactly!
The scrooge-generated scala classes look like this:
```
case object OneOff extends ophan.thrift.event.PaymentFrequency {
    val value: Int = 1
    val name: String = "OneOff"
    val originalName: String = "ONE_OFF"
    val annotations: immutable$Map[String, String] = immutable$Map.empty
  }
```

So wherever we were using `name` for GA, I've had to camel case, because the new scala model uses e.g. `ONE_OFF`